pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -592,4 +592,135 @@ func TestInt_Tasks(t *testing.T) {
+
+		middleTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		middleTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(middleTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootTask.ID()}))
+
+		id := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		task := createTaskWithRequest(t, sdk.NewCreateTaskRequest(id, sql).WithAfter([]sdk.SchemaObjectIdentifier{middleTask.ID()}))
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(middleTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(middleTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
+			require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(rootTask.ID()).WithSuspend(sdk.Bool(true))))
+		})
+
+		tasksToResume, err := client.Tasks.SuspendRootTasks(ctx, task.ID(), task.ID())
+		require.NoError(t, err)
+		require.NotEmpty(t, tasksToResume)
+		require.Contains(t, tasksToResume, rootTask.ID())
+
+		rootTaskStatus, err := client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateSuspended, rootTaskStatus.State)
+
+		middleTaskStatus, err := client.Tasks.ShowByID(ctx, middleTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, middleTaskStatus.State)
+
+		require.NoError(t, client.Tasks.ResumeTasks(ctx, tasksToResume))
+
+		rootTaskStatus, err = client.Tasks.ShowByID(ctx, rootTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, rootTaskStatus.State)
+
+		middleTaskStatus, err = client.Tasks.ShowByID(ctx, middleTask.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.TaskStateStarted, middleTaskStatus.State)
+	})
+
+	t.Run("suspend root tasks temporarily with three sequentially connected tasks - middle in DAG", func(t *testing.T) {
+		rootTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		rootTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(rootTaskId, sql).WithSchedule(sdk.String("60 minutes")))
+
+		middleTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		middleTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(middleTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{rootTask.ID()}))
+
+		childTaskId := sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.String())
+		childTask := createTaskWithRequest(t, sdk.NewCreateTaskRequest(childTaskId, sql).WithAfter([]sdk.SchemaObjectIdentifier{middleTask.ID()}))
+
+		require.NoError(t, client.Tasks.Alter(ctx, sdk.NewAlterTaskRequest(childTask.ID()).WithResume(sdk.Bool(true))))
+		t.Cleanup(func() {
```

Comment:
Why do we need to resume them in cleanup?

Reply:
It's necessary because dropping tasks in the cleanup is modifying the task graph. To resolve those errors we can either modify the task setup first to create tasks and then alter them to create connections (creating them in a different order will make t.Cleanup's run in the right order, but we would have to add alters to add "after's" to create graph) or leave it as is and create them in the "wrong" order with after's specified during create and pausing in t.Cleanup's to not error during calling DROP TASK X.